src/acl/Options.cc

```diff
@@ -250,32 +237,29 @@ Acl::NoOptions()
     return none;
 }
 
-const Acl::ParameterFlags &
-Acl::NoFlags()
+const Acl::BooleanOption &
+Acl::CaseSensitivityOption()
 {
-    static const ParameterFlags none;
-    return none;
+    static const BooleanOption MyOption("-i", "+i");
+    return MyOption;
 }
 
 std::ostream &
 operator <<(std::ostream &os, const Acl::Option &option)
 {
-    if (option.valued()) {
-        os << '=';
-        option.print(os);
-    }
+    option.print(os);
     return os;
 }
 
 std::ostream &
 operator <<(std::ostream &os, const Acl::Options &options)
 {
-    for (const auto pos: options) {
-        assert(pos.second);
-        const auto &option = *pos.second;
-        if (option.configured())
-            os << pos.first << option;
+    for (const auto option: options) {
+        if (option != options.front())
+            os << ' ';
```

Comment:
Does not this print extra spaces when options are supported but not actually configured?
https://github.com/squid-cache/squid/blob/235f0f1e4d10e99456bb5f85ec22ba9546fcb714/src/acl/Options.h#L181-L189

Reply:
You are right, I forgot about his condition in print(). I see two options:
* Adjust this approach, replacing configured() with Must(configured()) in print() and moving this check into operator <<() itself.
* Revert these changes and just add the leading ' ' into print().

Reply:
> Adjust this approach, replacing configured() with Must(configured()) in print() and moving this check into operator <<() itself.

FWIW, I kind of discussed and rejected that idea earlier, in https://github.com/measurement-factory/squid/commit/017d298996dbd462395ee77c7d140d4277922b7c

> Revert these changes and just add the leading ' ' into print().

Yes, that is what I tried to suggest at https://github.com/squid-cache/squid/pull/960#discussion_r782205553. If you try this, please adjust the callers to avoid extra spaces if that is not too difficult.

Reply:
Done.

>  adjust the callers to avoid extra spaces 

The only caller is ACL::dumpOptions(), which does not need adjusting.

Reply:
> The only caller is ACL::dumpOptions(), which does not need adjusting.

```C++
storeAppendPrintf(entry, "%s %s %s ",
                  name,
                  ae->name,
                  ae->typeString());
SBufList tail;
tail.splice(tail.end(), ae->dumpOptions());
tail.splice(tail.end(), ae->dump()); // ACL parameters
dump_SBufList(entry, tail);
```

I was thinking about the dump_acl() code shown above. Something still feels odd, even after your fixes: In dump_acl(), storeAppendPrintf() adds a trailing space, anticipating dumping options (and/or parameters) that did not add a leading space. Now dumpOptions() adds a leading space. I would expect the outcome to be an extra space (when some options are enabled).

Needless to say, simply removing that trailing space from storeAppendPrintf() would not work because that would break case 2 below.

We have at least these cases to worry about:

1. ACL without options and without parameters (rare but currently not impossible IIRC)
2. ACL without options but with parameters
3. ACL with options but without parameters
4. ACL with options and with parameters

I realize that fixing any of the above cases is outside this PR scope, but since all these changes were triggered by the out-of-scope "take the opportunity" change [request](https://github.com/squid-cache/squid/pull/960/files#r781809000), it is difficult to draw the line -- where do we stop? I am afraid that, right now, we have made things worse in case 3 (when there is only one option), improved case 3 (when there are multiple options), and we do not know whether there are any cases that are still broken...

If my suspicions are correct, then I recommend stopping here: Sprinkling our code with random space-adding snippets is not the right way towards a proper fix. Squid is simply missing the right API to dump directive parameters. That API cannot rely on std::ostream alone. Developing that API is clearly outside this PR scope! With that thought, I will resolve this change request.